evolver/hardware/demo.py

```diff
@@ -31,5 +31,12 @@ def get(self):
 
 
 class NoOpEffectorDriver(EffectorDriver):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.aborted = False
+
     def commit(self):
         self.comitted = copy(self.proposal)
+
+    def abort(self):
+        self.aborted = True
```

Comment:
this is the demo, the attr is specifically for testing

Reply:
I think this would be misleading to someone interpreting the "demo" as something to mimic in their implementations, which they shouldn't.

This would be better off in the test code itself, e.g., via a monkeypatch or something similar.